### PR TITLE
test: Add test for sampled transaction in hub

### PR DESCRIPTION
**Pull request by philipphofmann**
## :scroll: Description

#skip-changelog

## :bulb: Motivation and Context

I found this while implementing client reports.

## :green_heart: How did you test it?
Unit tests.

## :pencil: Checklist

- [x] I reviewed the submitted code
- [x] I added tests to verify the changes
- [x] I updated the docs if needed
- [x] Review from the native team if needed
- [x] No breaking changes

### CODE REVIEW

File: Sources/Sentry/SentryHub.m [hunk 1]
```diff
@@ -240,8 +240,10 @@ - (SentryId *)captureTransaction:(SentryTransaction *)transaction
                        withScope:(SentryScope *)scope
          additionalEnvelopeItems:(NSArray<SentryEnvelopeItem *> *)additionalEnvelopeItems
 {
-    if (transaction.trace.context.sampled != kSentrySampleDecisionYes)
+    if (transaction.trace.context.sampled == kSentrySampleDecisionYes) {
         return SentryId.empty;
+    }
```
[the remark this answers is not in the document]
Comment (philipphofmann) [reply Sources/Sentry/SentryHub.m:245]: Yes, exactly. Yes means it is sampled, see https://develop.sentry.dev/sdk/performance#header-sentry-trace

[the remark this answers is not in the document]
Comment (brustolin) [reply Sources/Sentry/SentryHub.m:245]: But you inverted the logic. Now, if sampled == YES, return empty SentryId and do not capture the event

File: Sources/Sentry/SentryTracesSampler.m [hunk 2]
```diff
@@ -52,10 +52,10 @@ - (SentrySampleDecision)sample:(SentrySamplingContext *)context
     return kSentrySampleDecisionNo;
 }
 
-- (SentrySampleDecision)calcSample:(double)rate
+- (SentrySampleDecision)calcSample:(double)sampleRate
 {
-    double r = [self.random nextNumber];
-    return r <= rate ? kSentrySampleDecisionYes : kSentrySampleDecisionNo;
+    double randomValue = [self.random nextNumber];
+    return randomValue <= sampleRate ? kSentrySampleDecisionNo : kSentrySampleDecisionYes;
```
Comment (brustolin) [inline Sources/Sentry/SentryTracesSampler.m:58]: Inverted logic.
```suggestion
    double randomValue = [self.random nextNumber];
    return randomValue <= sampleRate ? kSentrySampleDecisionYes : kSentrySampleDecisionNo;
```

File: Sources/Sentry/SentryHub.m (hunk 1 above)
[the remark this answers is not in the document]
Comment (philipphofmann) [reply Sources/Sentry/SentryHub.m:245]: Yes, I inverted the logic twice.

[the remark this answers is not in the document]
Comment (brustolin) [reply Sources/Sentry/SentryHub.m:245]: I see, you inverted it here, and also in the `calcSample:` method. 
But now we are using the wrong naming for things, which I believe is confusing and may cause problem in the future. 

According to the docs and java code that you mentioned, YES is used when a transaction should be capture not discarded.

[the remark this answers is not in the document]
Comment (philipphofmann) [reply Sources/Sentry/SentryHub.m:245]: Ah ups. Was confused a bit. I'm going to change it back then.

[the remark this answers is not in the document]
Comment (philipphofmann) [reply Sources/Sentry/SentryHub.m:245]: This means I can close the PR I think.

**Comment by philipphofmann** [comment]
@brustolin, I basically undid the PR. I changed it to add a missing test though.

**Review by brustolin — APPROVED** [review]

**Pull request merged**